File: 2308-divide-array-into-equal-pairs/divide-array-into-equal-pairs.java
class Solution {
    public boolean divideArray(int[] nums) {
        HashMap<Integer,Integer>freq=new HashMap<>();
        for(int i:nums){
            freq.put(i,freq.getOrDefault(i,0)+1);
        }
        for(var i:freq.entrySet()){
            if((i.getValue())%2!=0){
                return false;
            }
        }
        return true;
    }
}

File: 2308-divide-array-into-equal-pairs/divide-array-into-equal-pairs.py
from collections import defaultdict
class Solution:
    def divideArray(self, nums: List[int]) -> bool:
        freq=defaultdict(int)
        for i in nums:
            freq[i]+=1
        for i,j in freq.items():
            if j%2!=0:
                return False
        return True